src/core/ready.js

```diff
@@ -5,12 +5,13 @@ define( [
 ], function( jQuery, document ) {
 
 // The deferred used on DOM ready
-var readyList;
+var readyList = jQuery.Deferred(),
+	readyPromise = readyList.promise();
 
 jQuery.fn.ready = function( fn ) {
 
 	// Add the callback
-	jQuery.ready.promise().done( fn );
+	readyPromise.done( fn );
```

Comment:
I think it's much less likely to depend on that than to depend on synchronous `then` on a resolved promise and yet we made that change. Aligning those behaviors only seems good to me.